### PR TITLE
Qchip refactor

### PR DESCRIPTION
**Pull request by KilDesu**
## PR Type

> What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## What's new?

> List the changes

- Migration to Svelte 5
- Followed MD3 API rather than Quasar's
- Some minor changes in files "unrelated" to QChip

Important to note that for the `input` design, I omitted the "selected" state like we have on the `filter` design.
I don't see any use case for that.

I sometimes find MD3 specs kind of inconsistent, which makes decisions and refactors a bit harder sometimes.

## Screenshots

> If needed, you can add screenshots here

image

## This pull request closes an issue

> Add `Closes`, `Fixes` or `Resolves` followed by `#xxx[,#xxx]` where "xxx" is the issue number

Related to #61

### CODE REVIEW

File: src/lib/css/mixins/_design.scss [hunk 1]
```diff
@@ -45,24 +46,25 @@
 }
 
 @mixin disabled {
-  cursor: not-allowed;
-  box-shadow: none;
+  cursor: not-allowed !important;
```
Comment (therealcoder1337) [inline src/lib/css/mixins/_design.scss:49]: Is `!important` really necessary? Also for the others following.

**Review by therealcoder1337 — APPROVED** [review]
Awesome work! Let me know when you considered the comments and I'll merge.

File: src/lib/css/mixins/_design.scss (hunk 1 above)
Comment (KilDesu) [reply src/lib/css/mixins/_design.scss:49]: I think so, those styles are important and the specificity is pretty low so easily overwritten

File: src/lib/components/avatar/QAvatar.scss [hunk 2]
```diff
@@ -38,18 +38,17 @@
...
+  "lg": 3,
+  "xl": 4,
 );
 
 .q-avatar {
   @each $size, $val in $sizes {
     &--#{$size} {
-      height: 2rem + ($val - 1) * 0.5rem;
-      width: 2rem + ($val - 1) * 0.5rem;
+      --size: calc(2rem + #{$val} * 0.5rem);
```
[the remark this answers is not in the document]
Comment (KilDesu) [reply src/lib/components/avatar/QAvatar.scss:51]: Not sure that's really necessary

[the remark this answers is not in the document]
Comment (therealcoder1337) [reply src/lib/components/avatar/QAvatar.scss:51]: In the future, we should be clear with such things. No problem now.

File: src/lib/css/mixins/_design.scss (hunk 1 above)
Comment (therealcoder1337) [reply src/lib/css/mixins/_design.scss:49]: It's kind of a "last resort". Usually you should not use `!important`.

**Pull request merged**